app/models/agents/weather_agent.rb

```diff
@@ -100,14 +107,27 @@ def validate_options
     end
 
     def wunderground
-      Wunderground.new(interpolated['api_key']).forecast_for(location)['forecast']['simpleforecast']['forecastday'] if key_setup?
+      if key_setup?
+        forecast = Wunderground.new(interpolated['api_key'], language: language.upcase).forecast_for(location)
+
+        merged = Hash.new
+        forecast['forecast']['simpleforecast']['forecastday'].each{|daily| merged[daily['period']] = daily}
+
+        txt_forecast = Hash.new
+        merged.each_key do |day|
+            index = forecast['forecast']['txt_forecast']['forecastday'].find_index{|x| x['period'] == day}
```

Comment:
would it be possible to use [Enumerable#find](http://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-find) instead of getting the index and then accessing it later?